pyquil/experiment/_setting.py

```diff
@@ -66,112 +66,111 @@ class TensorProductState:
     states.
     """
 
-    states: Tuple[_OneQState]
+    states: List[_OneQState]
 
-    def __init__(self, states=None):
+    def __init__(self, states: Optional[List[_OneQState]] = None):
```

Comment:
Since you're already copying the `states` below, you could probably change this from `List` to `Sequence` or `Iterable`.